src/runtime/virtcontainers/factory/template/template_test.go

```diff
@@ -131,5 +131,5 @@ func TestTemplateFactory(t *testing.T) {
 
 	// expect tt.statePath not exist, if exist, it means this case failed.
 	_, err = os.Stat(tt.statePath)
-	assert.Nil(err)
+	assert.True(os.IsNotExist(err))
```

Comment:
@jodh-intel @Amulyam24 

I think this is depending on the environment. In GitHub actions, there will be a `unlinkat /tmp/vc/mockfs: device or resource busy` error, but in my locally Ubuntu(I think many other OSs are the same) it can be deleted successfully even the directory is occupied by other process/terminal.

So this assertion is environment-dependent. I think we can:
- Delete it
- Check case by case using this:

https://github.com/kata-containers/kata-containers/blob/72d1a04cf19c0eb6c709ebb0e2a49221c3a45151/src/runtime/pkg/katatestutils/utils.go#L348-L351

Reply:
@liubin According to @Amulyam24, the umount action is taking time to complete. So not sure if this has to do with the environment. Perhaps, we can check for the mounts until a timeout and then perform an os.Stat() on `tt.statePath`

Reply:
@amshinde @liubin, sorry about the delay(returning from a vacation). I tried with a sleep and this only works when the timeout is of 10 seconds duration. I'm not sure if this is feasible. PTAL.

Reply:
@Amulyam24 - I think @amshinde was suggesting we get the code to "poll" periodically, maybe up to, say, 60 seconds. Once the mounts have disappeared, we can then stat the path.

Reply:
@jodh-intel @amshinde, PTAL.